packages/core/test/merkle-distributor/MerkleDistributor.js

```diff
@@ -442,102 +456,95 @@ contract("MerkleDistributor.js", function(accounts) {
-          {
-            windowIndex: windowIndex + 1,
-            account: leaf2.account,
-            accountIndex: leaf2.accountIndex,
-            amount: leaf2.amount,
-            merkleProof: merkleTrees[1].getProof(leaf2.leaf)
-          }
-        ];
-        await merkleDistributor.claimMulti(claims, rewardToken.address, leaf1.account, { from: rando });
-
-        // Account 0 should have gained claimed amount from both leaves.
-        const batchedClaimAmount = toBN(leaf1.amount).add(toBN(leaf2.amount));
-        assert.equal(
-          (await rewardToken.balanceOf(leaf1.account)).toString(),
-          accountBalanceBefore.add(batchedClaimAmount).toString()
-        );
+        for (let i = 0; i < rewardRecipientAddresses.length; i++) {
+          // Trees 0 and 1 payout in rewardToken.
+          const expectedPayoutRewardToken = toBN(rewardLeafs[0][i].amount).add(toBN(rewardLeafs[1][i].amount));
+          // Trees 2 payout in altRewardToken
+          const expectedPayoutAltRewardToken = toBN(rewardLeafs[2][i].amount);
+
+          const account = rewardRecipientAddresses[i];
+          assert.equal(
+            balancesRewardToken[i].add(expectedPayoutRewardToken).toString(),
+            (await rewardTokens[0].balanceOf(account)).toString()
+          );
+          assert.equal(
+            balancesAltRewardToken[i].add(expectedPayoutAltRewardToken).toString(),
+            (await rewardTokens[1].balanceOf(account)).toString()
+          );
+        }
 
         // One Claimed event should have been emitted for each batched claim.
         const claimedEvents = await merkleDistributor.getPastEvents("Claimed");
-        assert.equal(claimedEvents.length, claims.length);
+        assert.equal(claimedEvents.length, rewardRecipientAddresses.length * 3);
       });
-      it("One tree: Can only batch claim for one account", async function() {
-        // Leaf 2 is for account[1], can't batch claim for two different accounts.
-        const leaf1 = rewardLeafs[0][0];
-        const leaf2 = rewardLeafs[0][1];
-
-        const invalidClaims = [
-          {
-            windowIndex: windowIndex,
-            account: leaf1.account,
-            accountIndex: leaf1.accountIndex,
-            amount: leaf1.amount,
-            merkleProof: merkleTrees[0].getProof(leaf1.leaf)
-          },
-          {
-            windowIndex: windowIndex,
-            account: leaf2.account,
-            accountIndex: leaf2.accountIndex,
-            amount: leaf2.amount,
-            merkleProof: merkleTrees[0].getProof(leaf2.leaf)
-          }
-        ];
-
-        // Show that verifyProof works for claims individually.
-        for (let claim of invalidClaims) {
-          assert.isTrue(await merkleDistributor.verifyClaim(claim));
+      it("gas", async function() {
+        const txn = await merkleDistributor.claimMulti(batchedClaims, rewardRecipientAddresses, rewardTokenAddresses);
+        assert.equal(txn.receipt.gasUsed, 1145721);
```

Comment:
Note that `claimMulti` on this trivial tree actually introduces a 40,000 gas increase compared to #2672, but on the larger 100k leaf tree we see gas savings for this implementation.